### ISSUE TITLE
no place to fill in deployment when using Azure openai

### ISSUE THREAD

**Issue by allanwakes** [Maplemx/Agently#105]
I tried to use Agently with Azure openai, but I found that there is no place to fill in deployment when initializing agent.
Take llamaindex for example:

```
llm = AzureOpenAI(  
    model="gpt-3.5-turbo-16k",  
    deployment_name="xxx",  
    api_key=os.getenv("AZURE_KEY"),  
    azure_endpoint="https://xxx.openai.azure.com/",  
    api_version="xxxx-xx-xx",  
    # temperature=0.6  
)  
```
Thanks.

**Comment by Maplemx** [comment]
try options

.set_settings("model.AzureOpenAI.options", { "deployment_name": "xxx" })

**Comment by allanwakes** [comment]
> try options
> 
> `.set_settings("model.AzureOpenAI.options", { "deployment_name": "xxx" })`

I tried with this options, but an exception pops out
`Exception: [Agent Request] Error: AsyncCompletions.create() got an unexpected keyword argument 'deployment_name'`

**Comment by Maplemx** [comment]
I see, I will check and fix this

**Comment by duoduo369** [comment]
try this，i checked source code, seems need use `"model.AzureOpenAI.options", {"model": model_name}` config

```
import Agently

agent_factory = Agently.AgentFactory()
# azure config
# https://github.com/Maplemx/Agently/blob/4e0f46b4d3c43113fad64a5231373e0d195f34df/Agently/plugins/request/AzureOpenAI.pyz#L36
model_name = os.environ['OPENAI_DEPLOYMENT_NAME']
api_key = os.environ['OPENAI_API_KEY']
azure_endpoint = os.environ['OPENAI_API_BASE']
api_version = os.environ['OPENAI_API_VERSION']
logger.info(model_name)
agent_factory.set_settings("current_model", "AzureOpenAI")\
    .set_settings("model.AzureOpenAI.options", {"model": model_name})\
    .set_settings("model.AzureOpenAI.auth.api_key", api_key)\
    .set_settings("model.AzureOpenAI.auth.azure_endpoint", azure_endpoint)\
    .set_settings("model.AzureOpenAI.auth.api_version", api_version)
```

![screenshot-20240614-095733](https://github.com/Maplemx/Agently/assets/6789022/7c14113d-0679-4ffb-be12-bf4e9f017584)

**Comment by Maplemx** [comment]
fixed in https://github.com/Maplemx/Agently/commit/4b97e3180cc35d57e0685eca6b35f58dbac72629

will be published in version 3.3.1.1

how to set:

```python
.set_settings("current_model", "AzureOpenAI")
        .set_settings("model.AzureOpenAI.auth", {
            "api_key": "xxxxx",
            "api_version": "2024-02-01",
            "azure_endpoint": "https://xxx.openai.azure.com/",
            "azure_deployment": "xxx",
        })
```

please confirm param names by reading Microsoft official document or reading azure python file in OpenAI package:

```python
class AzureOpenAI(BaseAzureClient[httpx.Client, Stream[Any]], OpenAI):
    @overload
    def __init__(
        self,
        *,
        azure_endpoint: str,
        azure_deployment: str | None = None,
        api_version: str | None = None,
        api_key: str | None = None,
        azure_ad_token: str | None = None,
        azure_ad_token_provider: AzureADTokenProvider | None = None,
        organization: str | None = None,
        timeout: float | Timeout | None | NotGiven = NOT_GIVEN,
        max_retries: int = DEFAULT_MAX_RETRIES,
        default_headers: Mapping[str, str] | None = None,
        default_query: Mapping[str, object] | None = None,
        http_client: httpx.Client | None = None,
        _strict_response_validation: bool = False,
    ) -> None:
        ...
```

**Comment by Maplemx** [comment]
published in version 3.3.1.1
